### ISSUE TITLE
path and environment bug

### ISSUE THREAD

**Issue by smuckola** [bluehost/wp-tools#13]
Hello and thank you for this tool.  It looks like the ultimate wp-cli management tool but I can't use it so far. All my sites are presently owned by apache:apache but even when I migrate to a user-per-site model, most of those users will still not have home directories and environments. Are those actually required? Should they really be? So it looks like I'm hitting the bug mentioned on the main page.  Also, 'strace' seems to show that wp-tools is stripping the leading / off of the pathnames if I'm reading it correctly, right?

[dtm@v ~]$ sudo -u apache /usr/local/bin/wp-tools --min-freemem=200000 --max-size=10000000 --path=/var/www/smuckola.org/w backup --backupdir=/tmp/backup --wp-cli=/usr/local/sbin/wp --username=apache
Error: This does not seem to be a WordPress install.
Pass --path=`path/to/wordpress` or run `wp core download`.
Unable to backup: Open pid file failed: Permission denied at /usr/local/bin/wp-tools line 491.
[dtm@v ~]$ 

Do you have any scripts that manage wp-tools, like something that would go in cron or systemd and run daily for a list of pathnames?

Are you still maintaining this app? Thanks.

**Comment by voldemortensen** [comment]
Home directories are not technically required, but this was definitely written with them existing as an assumption. Most *nix systems will use something like `/var/empty`, `/dev/null`, or `nonexistent` as the homedir for users that don't have a homedir. As long as the homedir in `/etc/passwd` cannot be `chdir`'d to by the user, it will stay in working directory of the script. `chdir` in perl just returns false if it fails.

The leading `/` are stripped, so as long as the specified path is relative to the working dir of the running script, it should work.

Sometime like:
`0 5 * * * cd /var/www/smuckola.org/w && /usr/bin/sudo -u /usr/local/bin/wp-tools --min-freemem=200000 --max-size=10000000 --path=. backup --backupdir=/tmp/backup --wp-cli=/usr/local/sbin/wp --username=apache >/dev/null 2>&1`
should work (I haven't tested that in any way). You'd just need to have it for every path.

Alternatively, something like
```
#!/bin/bash
PATH=/bin:/usr/bin/:/usr/local/bin:/sbin

sites=(/var/www/smuckola.org/w/ /var/www/anothersite/)

for site in ${sites[*]}; do
    cd $site
    su - apache -c wp-tools --min-freemem=200000 --max-size=10000000 --path=. backup --backupdir=/tmp/backup --wp-cli=/usr/local/sbin/wp --username=apache >/dev/null 2>&1
done
```
could work. Again, I haven't tested that at all.

> Are you still maintaining this app? Thanks.

I'm no longer with Bluehost, so I can't answer that particular question.

**Comment by chazmcgarvey** [comment]
Hey @smuckola,

Thanks for your interest. We wrote wp-tools in order to help people who manage a lot of really old WordPress sites get those sites updated to a secure version. We have already used it to update millions of websites, and we hope others have already or will continue to find it useful for that purpose.

If that's what you're looking for, then wp-tools might help you. But be aware that there are known bugs and usability issues (including the one mentioned in the README) that we would like to get fixed, and there hasn't been much active development on wp-tools because I believe it is feature-complete and has mostly served its purpose for most people.

The killer feature of wp-tools has been the ability to safely upgrade really old WordPress sites in a safe way (with some testing to make sure the sites still work after the upgrade). But for sites that are not ancient, we generally recommend turning on [WordPress automatic background updates](https://codex.wordpress.org/Configuring_Automatic_Background_Updates). If greater control is needed, then a much simpler wrapper around WP-CLI could be written; it may not have all the bells and whistles of wp-tools, but if you don't need those features then your site management might end up being simpler and faster without wp-tools.

That said, our original goal hasn't changed; we really want wp-tools to be useful for people upgrading old sites so that we all can have a more secure world wide web. Although we're not aiming to add new features, we do still want to fix bugs and help keep wp-tools useful for its main purpose. If you do end up deciding to integrate wp-tools into your site management, that's great and I'll try to help by providing support. If you make improvements to our code, I'd love to receive patches.
